examples/tutorials/self-paced-training/part-2_federated_learning_system/chapter-3_federated_computing_platform/03.0_introduction/introduction.ipynb

```diff
@@ -1,11 +1,169 @@
+    "\n",
+    "### Executor\n",
+    "\n",
+    "The Executor is the object that defines the logic to execute on the client side. It handles the tasks defined by the Controller and responds back to the task requests.\n",
+    "\n",
+    "The interaction between the Controller and Executor can be found in the following picture:\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "abe8109c",
+   "metadata": {},
+   "source": [
+    "### Shareable \n",
+    "\n",
+    "A [Shareable](https://nvflare.readthedocs.io/en/main/programming_guide/shareable.html) object represents communication between the server and client. Technically, a Shareable object is implemented as a Python dict. This dict contains two kinds of information:\n",
+    "* Header \n",
+    "    * Peer Properties\n",
+    "    * Cookie \n",
+    "    * Return code\n",
+    "* Content\n",
+    "\n",
+    "In other words, a Shareable is nothing but a dictionary with some metadata information.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "51e9e4fa",
+   "metadata": {},
+   "source": [
+    "The Controller and Executor exchange Shareable\n",
+    "\n",
+    "<img src=\"controller_executor_no_filter.png\" alt=\"Controller and executor\" width=\"700\" height=\"400\">\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "187ed5ee",
+   "metadata": {},
+   "source": [
+    "### Filters\n",
+    "\n",
+    "NVIDIA FLARE also introduces a filtering mechanism to allow users to limit the input & outputs. Filters in NVIDIA FLARE are a way to transform the Shareable object between the communicating parties. A [Filter](https://nvflare.readthedocs.io/en/main/programming_guide/filters.html) can be used to provide additional processing to shareable data before sending or after receiving from the peer.\n",
+    "\n",
+    "<img src=\"controller_worker_flow.png\" alt=\"Controller and executor with filters\" width=\"700\" height=\"400\">\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fd3c8200",
+   "metadata": {},
+   "source": [
+    "### FLComponent\n",
+    "\n",
+    "NVIDIA FLARE is built with components. FLComponent is the building block of all components (Base Class). Controller, Executor, Filter, and Shareable are all types of FLComponent.\n",
+    "\n",
+    "The core property of FLComponent is event support. FLComponent is able to fire and receive events, enabling the FLARE system to be an event-driven, pluggable system.\n",
+    "\n",
+    "### FLContext\n",
+    "One of the most important features of NVIDIA FLARE is nvflare.apis.fl_context to pass data between the FL components. FLContext is available to every method of all FLComponent types (Controller, Aggregator, Filter, Executor).\n",
```

Comment:
`nvflare.apis.fl_context`

Reply:
just want to highlight the "nvflare.apis.fl_context" by using single quotation marks